### ISSUE TITLE
iH - add JSON output

### ISSUE THREAD

**Issue by geekscrapy** [rizinorg/rizin#1978]
It would be great to get JSON output of the iH command

I think it was originally raised here: https://github.com/radareorg/radare2/pull/14515

**Comment by wargio** [comment]
This is kinda hard to implement to be as generic as possible outside to implement this per bin format.

**Comment by ret2libc** [comment]
Yes indeed I am not sure how much benefit there would be in such output because you would need to know what type of file you are dealng with to even know which fields to expect in the JSON.. Maybe it's still useful, but not the best scenario. Other ideas?

**Comment by XVilka** [comment]
You can wrap the iH output in the "{ format: elf, header: { }}" or
something like that.

**Comment by ret2libc** [comment]
still, it doesn't help you much because then you (as the user of the API/command) have to know that elf has "type", "machine", "version", "entrypoint", etc. while pe has something else. Which might be alright, not saying it isn't, but it should be clear at least.

**Comment by wargio** [comment]
maybe we can output in json format `pf`
